test/Numerics/Mesh/topology.jl

```diff
@@ -94,6 +94,17 @@ end
     @test length(g) == 101
     @test g[1] == 10
     @test g[end] == 20
+
+    g = grid1d(
+        10f0,
+        20f0,
+        DoubleSidedSingleExponentialStretching(2.5f0),
+        elemsize = 0.1,
+    )
+    @test eltype(g) == Float32
+    @test length(g) == 101
+    @test g[1] == 10
+    @test g[end] == 20
```

Comment:
Would it make sense to check more than the end points to see if the stretching was applied correctly?

Reply:
I could calculate the expected result and have it be compared to the function's output here. Would that be a better test ?

Reply:
I've added this.

Reply:
Seems reasonable.  You man want to use `≈` or `isapprox` instead of `==` to allow for differences due to floating point arithmetic.